### ISSUE TITLE
`UnnecessaryQualifier` emits false positive warning when using Guice `@AssistedInject`

### ISSUE THREAD

**Issue by jnehlmeier** [google/error-prone#5273]
The code below triggers an incorrect warning on `@InjectCurrentTimestamp`.

```java

class Test {

  interface Factory {
    Test create(String paramFromFactory);
    Test create(Integer paramFromFactory);
  }

  @AssistedInject
  Test(@Assisted String paramFromFactory, @InjectCurrentTimestamp now) {
  }

  @AssistedInject
  Test(@Assisted Integer paramFromFactory, @InjectCurrentTimestamp now) {
  }

}

```
